Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -3754,44 +3755,157 @@ JS_DEFINE_NATIVE_FUNCTION(@class_name@::@attribute.getter_callback@)
+                }
+
+                // 4. If the reflected IDL attribute is limited to only non-negative numbers, then return −1.
+                // 5. Return 0.
                 attribute_generator.append(R"~~~(
     i32 retval = 0;
+    bool did_set_value = false;
     auto content_attribute_value = impl->get_attribute("@attribute.reflect_name@"_fly_string);
     if (content_attribute_value.has_value()) {
-        auto maybe_parsed_value = Web::HTML::parse_integer(*content_attribute_value);
-        if (maybe_parsed_value.has_value())
-            retval = *maybe_parsed_value;
+        if (@positive@) {
+            auto maybe_parsed_value = Web::HTML::parse_non_negative_integer(*content_attribute_value);
+
+            if (maybe_parsed_value.has_value()) {
+                auto parsed_value = maybe_parsed_value.value();
+
+                if (parsed_value <= NumericLimits<WebIDL::Long>::max()) {
+                    retval = parsed_value;
+                    did_set_value = true;
+                }
+            }
+        } else {
+            auto maybe_parsed_value = Web::HTML::parse_integer(*content_attribute_value);
+            
+            if (maybe_parsed_value.has_value()) {
+                auto parsed_value = maybe_parsed_value.value();
+
+                if (parsed_value <= NumericLimits<WebIDL::Long>::max() && parsed_value >= NumericLimits<WebIDL::Long>::min()) {
+                    retval = parsed_value;
+                    did_set_value = true;
+                }
+            }
+        }
+    } 
+    
+    if (@has_default_value@ && !did_set_value) {
+        retval = @default_value@;
+        did_set_value = true;
+    }
+    
+    if (@positive@ && !did_set_value) {
+        retval = -1;
+        did_set_value = true;
+    } 
+    
+    if (!did_set_value) {
+        retval = 0;
     }
 )~~~");
             }
             // If a reflected IDL attribute has the type unsigned long,
-            // FIXME: optionally limited to only positive numbers, limited to only positive numbers with fallback, or clamped to the range [clampedMin, clampedMax], and optionally with a default value defaultValue:
+            // optionally limited to only positive numbers, limited to only positive numbers with fallback, or clamped to the range [clampedMin, clampedMax], and optionally with a default value defaultValue:
             else if (attribute.type->name() == "unsigned long") {
                 // The getter steps are:
                 // 1. Let contentAttributeValue be the result of running this's get the content attribute.
                 // 2. Let minimum be 0.
-                // FIXME: 3. If the reflected IDL attribute is limited to only positive numbers or limited to only positive numbers with fallback, then set minimum to 1.
-                // FIXME: 4. If the reflected IDL attribute is clamped to the range, then set minimum to clampedMin.
+                u32 minimum = 0;
+                u32 maximum = 2147483647;
```

Comment:
My preference would be to move this declaration to step 5, where it's introduced with this value.